server/datastore/mysql/policies.go

```diff
@@ -1663,3 +1663,38 @@ func (ds *Datastore) GetTeamHostsPolicyMemberships(
 
 	return hosts, nil
 }
+
+// GetPoliciesBySoftwareTitleID returns the policies that are associated with a set of software titles.
+func (ds *Datastore) getPoliciesBySoftwareTitleIDs(
+	ctx context.Context,
+	softwareTitleIDs []uint,
+	teamID *uint,
+) ([]fleet.AutomaticInstallPolicy, error) {
+	query := `
+	SELECT
+		p.id AS id,
+		p.name AS name,
+		st.id AS software_title_id
+	FROM policies p
+	JOIN software_installers si ON p.software_installer_id = si.id
+	JOIN software_titles st ON si.title_id = st.id
+	WHERE st.id IN (?) AND p.team_id = ?
+	GROUP BY software_title_id, id;
```

Comment:
Do you really need a `GROUP BY`? You want the whole list of policies with the matching software title id, so unless I'm missing something I don't think there's a need for that group by?

Reply:
@mna ran the test locally w/o group by and it worked.